tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -238,3 +271,174 @@ def get_retention_from_backup_policy(self, retention_policy):
         if actual_duration_units.iso8601_symbol != self.retention_period_units.iso8601_symbol:
             return None
         return actual_duration
+
+
+@six.add_metaclass(abc.ABCMeta)
+class BackupRetentionPolicyAction(AzureBaseAction):
+
+    def __init__(self, operations_property, *args, **kwargs):
+        super(BackupRetentionPolicyAction, self).__init__(*args, **kwargs)
+        self.operations_property = operations_property
+
+    def _process_resource(self, database):
+        client = self.manager.get_client()
+        update_operation = getattr(client, self.operations_property).create_or_update
+
+        resource_group_name, server_name, database_name = \
+            BackupRetentionPolicyHelper.get_backup_retention_policy_context(database)
+        parameters = self.get_parameters_for_new_retention_policy(database)
+
+        update_operation(resource_group_name, server_name, database_name, parameters).result()
+
+    @abc.abstractmethod
+    def get_parameters_for_new_retention_policy(self, database):
+        raise NotImplementedError()
+
+
+@SqlDatabase.action_registry.register('update-short-term-backup-retention-policy')
+class ShortTermBackupRetentionPolicyAction(BackupRetentionPolicyAction):
+    """
+
+    Update the short term backup retention policy for a SQL Database.
+
+    :example: Update any SQL Database short term retentions to at least 7 days.
+
+    .. code-block:: yaml
+
+            policies:
+              - name: update-short-term-backup-retention-policy
+                resource: azure.sqldatabase
+                filters:
+                  - type: short-term-backup-retention-policy
+                    op: lt
+                    retention-period-days: 7
+                actions:
+                  - type: update-short-term-backup-retention-policy
+                    retention-period-days: 7
+
+    """
+
+    VALID_RETENTION_PERIOD_DAYS = [7, 14, 21, 28, 35]
+
+    schema = type_schema(
+        'update-short-term-backup-retention-policy',
+        rinherit=ShortTermBackupRetentionPolicyFilter.schema,
+        op=None
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(ShortTermBackupRetentionPolicyAction, self).__init__(
+            BackupRetentionPolicyHelper.SHORT_TERM_SQL_OPERATIONS, *args, **kwargs)
+        self.retention_period_days = self.data['retention-period-days']
+
+    def validate(self):
+        if self.retention_period_days not in \
+                ShortTermBackupRetentionPolicyAction.VALID_RETENTION_PERIOD_DAYS:
+            raise PolicyValidationError(
+                "Invalid retention-period-days: {}. Valid values are: {}".format(
+                    self.retention_period_days,
+                    ShortTermBackupRetentionPolicyAction.VALID_RETENTION_PERIOD_DAYS
+                )
+            )
+        return self
+
+    def get_parameters_for_new_retention_policy(self, database):
+        return self.retention_period_days
```

Comment:
I though parameters needs to be an Azure SDK object (or at least a dict, like we do in other places)

Reply:
in the case of short term retentions, the python SDK simply passes an `int` as the parameter. They don't call it "parameters" the same way as other SDK calls do either.

[this is the sdk call](https://docs.microsoft.com/en-us/python/api/azure-mgmt-sql/azure.mgmt.sql.operations.backup_short_term_retention_policies_operations.backupshorttermretentionpoliciesoperations?view=azure-python#create-or-update-resource-group-name--server-name--database-name--retention-days-none--custom-headers-none--raw-false--polling-true----operation-config-)